homeassistant/components/home_connect/switch.py

```diff
@@ -19,12 +22,55 @@
     BSH_POWER_ON,
     BSH_POWER_STATE,
     DOMAIN,
+    REFRIGERATION_DISPENSER,
+    REFRIGERATION_SUPERMODEFREEZER,
+    REFRIGERATION_SUPERMODEREFRIGERATOR,
 )
-from .entity import HomeConnectEntity
+from .entity import HomeConnectDevice, HomeConnectEntity
 
 _LOGGER = logging.getLogger(__name__)
 
 
+@dataclass(frozen=True)
+class HomeConnectSwitchEntityDescription(SwitchEntityDescription):
+    """Switch entity description."""
+
+    on_key: str | None = None
+    value_fn: Callable[[dict], bool] = lambda _: True
+    exist_fn: Callable[[HomeConnectDevice], bool] = lambda _: False
+
+
+SWITCHES: tuple[HomeConnectSwitchEntityDescription, ...] = (
+    HomeConnectSwitchEntityDescription(
+        key="Supermode Freezer",
+        on_key=REFRIGERATION_SUPERMODEFREEZER,
+        value_fn=lambda status: status.get(REFRIGERATION_SUPERMODEFREEZER, {}).get(
+            ATTR_VALUE, False
+        ),
+        exist_fn=lambda device: REFRIGERATION_SUPERMODEFREEZER
+        in device.appliance.status,
+    ),
+    HomeConnectSwitchEntityDescription(
+        key="Supermode Refrigerator",
+        on_key=REFRIGERATION_SUPERMODEREFRIGERATOR,
+        value_fn=lambda status: status.get(REFRIGERATION_SUPERMODEREFRIGERATOR, {}).get(
+            ATTR_VALUE, False
+        ),
+        exist_fn=lambda device: REFRIGERATION_SUPERMODEREFRIGERATOR
+        in device.appliance.status,
+    ),
+    HomeConnectSwitchEntityDescription(
+        key="Dispenser Enabled",
+        on_key=REFRIGERATION_DISPENSER,
+        translation_key="refrigeration_dispenser",
+        value_fn=lambda status: status.get(REFRIGERATION_DISPENSER, {}).get(
+            ATTR_VALUE, False
+        ),
```

Comment:
since all 3 do `.get(ATTR_VALUE, False)`, would it make sense to move that to the `HomeConnectSwitch`?

Reply:
maybe value_fn could be removed to `status_key` and you just pass `REFRIGERATION_SUPERMODEFREEZER` and have the entity do it